### PR TITLE
feat: reduce torch.compile warmup recompilations for Qwen3 MoE

### PR DESCRIPTION
**Pull request by kamil-kaczor**
## Summary

Add HPU-specific Qwen3 MoE model classes that reduce `torch.compile` warmup time by eliminating unnecessary recompilations.

## Changes

- **HpuQwen3MoeModel**: Use `torch.zeros_like(hidden_states)` instead of `residual=None` to avoid type guard failures at every decoder layer
- **HpuQwen3MoeSparseMoeBlock**: Handle 3D tensor reshaping and tuple returns from SharedFusedMoE  
- **HpuQwen3MoeDecoderLayer**: Optional `torch.compiler.disable()` graph break placement controlled by `VLLM_MOE_GRAPH_BREAK` env var (`inside`/`after`/`none`, default: `none`)
- **mark_unbacked**: Scope `mark_unbacked` on block_list dim 0 to `flatten_input` (MoE) models only to avoid size specialization
- **Config**: Register `VLLM_MOE_GRAPH_BREAK` in the config system
- **Docs**: Document `VLLM_MOE_GRAPH_BREAK` in env_variables.md

## Motivation

MoE models like Qwen3-30B-A3B suffer from excessive recompilations during `torch.compile` warmup due to:
1. `residual=None` at the first decoder layer causing type guard failures at every layer boundary
2. Block list tensor size specializations triggering recompilations

This PR reduces warmup recompilations by ~23% and guard failures by ~68% with the residual fix alone. The optional graph break (`VLLM_MOE_GRAPH_BREAK=true`) can further reduce warmup time at the cost of ~9% TPOT overhead at low concurrency.

## Testing

- Validated with Qwen3-30B-A3B (TP=2, EP) on Gaudi3
- Zero runtime recompiles confirmed
- `ruff check` and pre-commit hooks pass

### CODE REVIEW

File: vllm_gaudi/v1/worker/hpu_model_runner.py [hunk 1]
```diff
@@ -640,6 +640,13 @@ def forward(self, *args, **kwargs):
                                                                                input_ids.size(0), input_ids.size(1),
                                                                                input_ids.device, self.dtype,
                                                                                model_has_chunked_attention)
+        # Avoid 0/1 size specialization on block_list dim 0 for MoE models.
+        # Must be outside compiled regions — mark_unbacked is a forbidden callable.
+        if self.flatten_input:
+            attn_md = kwargs.get('attn_metadata')
+            if (attn_md is not None and getattr(attn_md, 'is_prompt', False)
+                    and getattr(attn_md, 'block_list', None) is not None):
+                torch._dynamo.decorators.mark_unbacked(attn_md.block_list, 0)
```
[the remark this answers is not in the document]
Comment (afierka-intel) [reply vllm_gaudi/v1/worker/hpu_model_runner.py]: This suggestion seems to be valid. @kamil-kaczor consider to apply it.
